pkg/operation/botanist/clusteridentity/clusteridentity.go

```diff
@@ -0,0 +1,97 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusteridentity
+
+import (
+	"context"
+	"fmt"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type shootClusterIdentity struct {
+	clusterIdentity *string
+	seedNamespace   string
+	namespace       string
+	name            string
+	uid             string
+
+	gardenClient client.Client
+
+	logger *logrus.Entry
+}
+
+// New creates new instance of Deployer for Shoot cluster identity
+func New(
+	clusterIdentity *string,
+	name, namespace, seedNamespace, uid string,
+	gardenClient client.Client,
+	logger *logrus.Entry,
+) component.Deployer {
+	return &shootClusterIdentity{
+		clusterIdentity: clusterIdentity,
+		name:            name,
+		namespace:       namespace,
+		uid:             uid,
+		seedNamespace:   seedNamespace,
+		gardenClient:    gardenClient,
+		logger:          logger,
+	}
+}
+
+func (c *shootClusterIdentity) Deploy(ctx context.Context) error {
+	if c.clusterIdentity == nil {
+		clusterIdentity, err := c.computeShootClusterIdentity(ctx)
+		if err != nil {
+			return err
+		}
+
+		shoot := &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      c.name,
+				Namespace: c.namespace,
+			},
+		}
+
+		patch := []byte(fmt.Sprintf(`{"status": {"clusterIdentity": "%s"}}`, clusterIdentity))
+		return c.gardenClient.Status().Patch(ctx, shoot, client.RawPatch(types.StrategicMergePatchType, patch))
+	}
+	return nil
+}
+
+// Destroy returns nil
+func (c *shootClusterIdentity) Destroy(ctx context.Context) error {
+	return nil
+}
+
+// computeShootClusterIdentity computes the identity of a Shoot cluster and returns it.
+func (c *shootClusterIdentity) computeShootClusterIdentity(ctx context.Context) (string, error) {
+	gardenClusterIdentity := &corev1.ConfigMap{}
+	if err := c.gardenClient.Get(ctx, kutil.Key(metav1.NamespaceSystem, v1beta1constants.ClusterIdentity), gardenClusterIdentity); err != nil {
+		return "", err
+	}
+
+	clusterIdentity := fmt.Sprintf("%s-%s-%s", c.seedNamespace, c.uid, gardenClusterIdentity.Data[v1beta1constants.ClusterIdentity])
```

Comment:
Why do we bother generating the Shoot cluster identity in this ways instead of just using an UID (perhaps the `c.uid` part), similarly to the Seed?

Reply:
@mandelsoft point here was that in the IaaS provider for example, the user will only see an UID that will not give any hint where this is comming from